aten/src/ATen/native/cpu/IndexKernel.cpp

```diff
@@ -526,6 +529,200 @@ void cpu_hflip_vec(at::TensorIterator& iter) {
+
+  int64_t i = 0;
+#ifdef CPU_CAPABILITY_AVX2
+
+  const auto vec_size = 256 / (8 * stride);
+
+  if (size > vec_size) {
+
+    // Example for size0=3
+    // Data: (1 2 3) (4 5 6) (7 8 9) (10 11 12) (13 14 15) (16 17 18) (19 20 21) (22 23 24) (25 26 27) (28 29 30) (31 32 33)
+    // load by 2 parts
+    // R = [ (1 2 3) (4 5 6) (7 8 9) (10 11 12) (13 14 15) (16 | (16 17 18) (19 20 21) (22 23 24) (25 26 27) (28 29 30) (31 ]
+    // flip(R) ->
+    // R = [ 31 (28 29 30) (25 26 27) (22 23 24) (19 20 21) (16 17 18) | 16 (13 14 15) (10 11 12) (7 8 9) (4 5 6) (1 2 3) ]
+    //
+    // Write in 2 parts
+    // Output pointer:                                                                                                       v
+    // - Init:
+    //                (X X X)  (X X X)    (X X X)    (X X X)    (X X X)    (X X X)    (X X X)    (X X X)    (X X X) (X X X) (X X X)
+    // 0) Move to initial position: output_ptr += (size0 - delta/2) * stride;
+    //                                                                                                                           v
+    //                (X X X)  (X X X)    (X X X)    (X X X)    (X X X)    (X X X)    (X X X)    (X X X)    (X X X) (X X X) (X X X)
+    // - In the loop:
+    // 1) Move to write position: output_ptr -= usable_vec_stride and write 1st block from output_ptr + usable_vec_half_stride
+    //                     v                                                    | ----> vec_size / 2 ----------------------------|
+    // Output part 1: (X X X)  (X X X)    (X X X)    (X X X)    (X X X)    (X X 16)   (13 14 15) (10 11 12) (7 8 9) (4 5 6) (1 2 3)
+    // 2) Write 2nd block from output_ptr:
+    //                     v ----> vec_size / 2 -----------------------------------|
+    // Output part 2: (X X 31) (28 29 30) (25 26 27) (22 23 24) (19 20 21) (16 17 18) (13 14 15) (10 11 12) (7 8 9) (4 5 6) (1 2 3)
+    //
+    // - After the loop:
+    // 3) Move to write position: output_ptr -= (size0 - delta/2) * stride;
+    //                 v
+    //                (X X 31) (28 29 30) (25 26 27) (22 23 24) (19 20 21) (16 17 18) (13 14 15) (10 11 12) (7 8 9) (4 5 6) (1 2 3)
+
+    // Generate shuffle mask for N channels, N in [2, 8]
+    char mdata[16] = {};
+    for (int k=0; k<16; k++) {
+      int j = k / size0 + 1;
+      int v = (j * size0 - 1) - (k % size0);
+      v = std::min(v, (int) 15);
+      mdata[k] = v;
+    }
+    __m256i mask = _mm256_set_epi8(
+      mdata[0], mdata[1], mdata[2], mdata[3],
+      mdata[4], mdata[5], mdata[6], mdata[7],
+      mdata[8], mdata[9], mdata[10], mdata[11],
+      mdata[12], mdata[13], mdata[14], mdata[15], // first 128-bit lane
+
+      mdata[0], mdata[1], mdata[2], mdata[3],
+      mdata[4], mdata[5], mdata[6], mdata[7],
+      mdata[8], mdata[9], mdata[10], mdata[11],
+      mdata[12], mdata[13], mdata[14], mdata[15] // second 128-bit lane
+    );
+
+    const auto usable_vec_size = 2 * ((vec_size / 2) / size0);
```

Comment:
NIT: This is a much more relevant quantity
```suggestion
    const auto usable_vec_size = 2 * ((vec_size / 2) / size0) * size0;
```